.github/workflows/cache-deps.yaml

```diff
@@ -0,0 +1,42 @@
+#################################################################################
+# Child pipeline to cache hoprnet dependencies
+#################################################################################
+name: Cache dependencies
+env:
+  RUST_BACKTRACE: "1"
+on:
+  workflow_call:
+    inputs:
+      branch:
+        required: true
+        type: string
+concurrency:
+  group: ${{ github.ref }}-cache-deps
+  cancel-in-progress: true
+jobs:
+  cargo:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.branch }}
+          persist-credentials: false
+      - name: Replace the crates.io registry with a private proxy registry
+        env:
+          PRIVATE_CARGO_REGISTRY_PROXY_URL: ${{ secrets.PRIVATE_CARGO_REGISTRY_PROXY_URL }}
```

Comment:
Is the `PRIVATE_CARGO_REGISTRY_PROXY_URL` actually defined in the repo secrets?

Reply:
@esterlus needs to do that